design-documents/testing/functional/bdd-like-mftf-output.md

```diff
@@ -0,0 +1,141 @@
+# Overview
+
+Magento Functional Testing Framework (MFTF) aims to generate valuable output for different cases:
+
+* **Debug** - understanding what happens _under the hood_, what are the names of steps and what selectors were used.
+  ```gherkin
+  [amOnAdminLoginPage] I am on Admin URL (Admin URL: "/god/admin")
+  [fillUsername] I fill username field with "god" (xpath: "#username")
+  [fillPassword] I fill password field with "79XedTOBjPiHFtLt" (xpath: "#login")
+  [clickOnSignIn] I click Login button (xpath: ".actions .action-primary")
+  [waitForPageload] I wait for page load (timeout: 30)
+  [seeAdminLoginUrl] I see Admin URL in current url (Admin URL: "/god/admin")
+  PASSED
+  ```
+
+* **Non-verbose** (default) - showing just informational messages.
+  ```gherkin
+  I am on Admin URL
+  I fill username field with "god"
+  I fill password field with "79XedTOBjPiHFtLt"
+  I click Login button
+  I wait for page load
+  I see Admin URL in current url
+  PASSED
+  ```
+
+* **Behavioral** - focused on behavior, not specific actions.
+  ```gherkin
+  I am logged in as Administrator (login: "god", password: "xyz123")
+  I am on Product Grid
+  I click Add Product button
+  I save screenshot
+  ```
+
+## PoC
+
+https://github.com/magento/magento2-functional-testing-framework/issues/364
+
+## Usage
+
+Output should be consistent across the outputs:
+- Allure Report
+- CLI Report
+- Documentation
+
+## Behavioral output
+
+The real value of such output is possibility to confront the Business Requirements with Test Scenario without any extra interpretation. The content should be obvious to non-technical person.
+
+Example of such is Sylius Framework: https://github.com/Sylius/Sylius/blob/48729b86334ba78d8f2011b9743389024c334d56/features/checkout/paying_for_order/paying_with_paypal_during_checkout.feature#L7-L24
+
+### User-defined Label Attribute
+
+Adding extra attribute to XML nodes that allows to define user-friendly label for element / action / action group.
+
+* Element
+ ```xml
+ <element name="password" type="input" selector="#login" label="Password Input"/>
```

Comment:
`<element>`s are (usually) named something intuitive for the test writer, so I'm not 100% sure the benefit of adding a separate verbosity to elements.

In your examples it's the difference between:
`I fill password` vs `I fill Password Input`

I'm not the biggest believer in BDD/Gherkin style test output but I understand it's a matter of preference, so I'd prefer keeping it as non-instrusive for developers (meaning I'd prefer utilizing the existing `type` attribute).